src/apns_connection.erl

```diff
@@ -467,21 +630,22 @@ get_device_path(DeviceId) ->
 add_authorization_header(Headers, Token) ->
   Headers#{apns_auth_token => <<"bearer ", Token/binary>>}.
 
--spec push(pid(), apns:device_id(), apns:headers(), notification(), integer()) ->
-  apns:stream_id().
-push(GunConn, DeviceId, HeadersMap, Notification, Timeout) ->
+-spec send_push(pid(), apns:device_id(), apns:headers(), notification()) ->
+  gun:stream_ref().
+send_push(GunPid, DeviceId, HeadersMap, Notification) ->
   Headers = get_headers(HeadersMap),
   Path = get_device_path(DeviceId),
-  StreamRef = gun:post(GunConn, Path, Headers, Notification),
-  case gun:await(GunConn, StreamRef, Timeout) of
-      {response, fin, Status, ResponseHeaders} ->
-        {Status, ResponseHeaders, no_body};
-      {response, nofin, Status, ResponseHeaders} ->
-        {ok, Body} = gun:await_body(GunConn, StreamRef, Timeout),
-        DecodedBody = jsx:decode(Body, [{return_maps, false}]),
-        {Status, ResponseHeaders, DecodedBody};
-      {error, timeout} -> timeout
-  end.
+  gun:post(GunPid, Path, Headers, Notification).
+
+  %% case gun:await(GunPid, StreamRef, Timeout) of
+  %%     {response, fin, Status, ResponseHeaders} ->
+  %%       {Status, ResponseHeaders, no_body};
+  %%     {response, nofin, Status, ResponseHeaders} ->
+  %%       {ok, Body} = gun:await_body(GunPid, StreamRef, Timeout),
+  %%       DecodedBody = jsx:decode(Body, [{return_maps, false}]),
+  %%       {Status, ResponseHeaders, DecodedBody};
+  %%     {error, timeout} -> timeout
+  %% end.
```

Comment:
Please don't leave commented out code. If it's not needed, remove it.

Reply:
fixed